src/DssSpell.sol

```diff
@@ -19,66 +19,48 @@ pragma solidity 0.6.12;
+    function bump(bytes32, uint256) external;
 }
 
 contract DssSpellAction is DssAction {
 
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
-    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/3e69d2d9403ad0db916a370be66b9b77970b9a4d/governance/votes/Executive%20vote%20-%20July%2016%2C%202021.md -q -O - 2> /dev/null)"
+    // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/6d2eaab1360395916b9b09c4137c5da6f2de9f1a/governance/votes/Executive%20vote%20-%20July%2017%2C%202021.md -q -O - 2> /dev/null)"
     string public constant override description =
-        "2021-07-16 MakerDAO Executive Spell | Hash: 0x3797ab3aade24e8dbc3c859932f2a4fde3acbef215447b133e1622c21813dfab";
+        "2021-07-17 MakerDAO Executive Spell | Hash: 0x887d27e20f47d0701d1eea14045bb5fdbea634401f7dc5aedd729af13a5c8ddc";
 
     // Turn off office hours
     function officeHours() public override returns (bool) {
         return false;
     }
 
+    uint256 constant WAD     = 10 ** 18;
     uint256 constant MILLION = 10 ** 6;
 
-    // Growth Core Unit
-    address constant GRO_MULTISIG        = 0x7800C137A645c07132886539217ce192b9F0528e;
-    // Content Production Core Unit
-    address constant MKT_MULTISIG        = 0xDCAF2C84e1154c8DdD3203880e5db965bfF09B60;
-    // GovAlpha Core Unit
-    address constant GOV_MULTISIG        = 0x01D26f8c5cC009868A4BF66E268c17B057fF7A73;
-    // Real-World Finance Core Unit
-    address constant RWF_MULTISIG        = 0x9e1585d9CA64243CE43D42f7dD7333190F66Ca09;
-    // Risk Core Unit
-    address constant RISK_CU_EOA         = 0xd98ef20520048a35EdA9A202137847A62120d2d9;
-    // Protocol Engineering Multisig
-    address constant PE_MULTISIG         = 0xe2c16c308b843eD02B09156388Cb240cEd58C01c;
-    // Oracles Core Unit (Operating)
-    address constant ORA_MULTISIG        = 0x2d09B7b95f3F312ba6dDfB77bA6971786c5b50Cf;
-    // Oracles Core Unit (Emergency Fund)
-    address constant ORA_ER_MULTISIG     = 0x53CCAA8E3beF14254041500aCC3f1D4edb5B6D24;
-
-    address constant PREV_SPELL = 0xEC782b5aC1f0Fc096Ad30950f3348670980f7FD3;
+    address constant MIP21_LIQUIDATION_ORACLE = 0x88f88Bb9E66241B73B84f3A6E197FbBa487b1E30;
 
     function actions() public override {
 
         // https://vote.makerdao.com/polling/QmUNouQ7?network=mainnet#poll-detail
+
+        bytes32 ilk = bytes32("RWA002-A");
+
         // Will also increase the global debt ceiling.
-        DssExecLib.increaseIlkDebtCeiling(bytes32("RWA002-A"), 15 * MILLION, true);
+        DssExecLib.increaseIlkDebtCeiling(ilk, 15 * MILLION, true);
+
+        // Must increase the price to enable DAI to be drawn--value corresponds to
+        // [ (debt ceiling) + (2 years interest at current rate) ] * mat, i.e.
+        // 20MM * 1.035^2 * 1.05 as a WAD
+        Bumpable(MIP21_LIQUIDATION_ORACLE).bump(ilk, 22_495_725 * WAD);
```

Comment:
Since the price is derived from the other values, and to help with future spells, maybe this should be calculated rather than hardcoded? Something like:

```solidity
uint256 constant THREE_PT_FIVE_PCT  = 1000000001090862085746321732;
uint256 constant SECONDS_IN_YEAR = 60 * 60 * 24 * 365;

// [ (debt ceiling) + (2 years interest at current rate) ] * mat
function calculateMip21Price(uint ceil, uint rate, uint mat) internal pure returns (uint) {
    return rmul(rmul(ceil * WAD, rpow(rate, SECONDS_IN_YEAR*2)), mat * 10**22);
}
```

```suggestion
        Bumpable(MIP21_LIQUIDATION_ORACLE).bump(ilk, calculateMip21Price(20 * MILLION, THREE_PT_FIVE_PCT, 10_500);
```

Reply:
We'll build that into a library or something. No need to do it here.

Reply:
Hi @offerijns, perhaps. I think this is a specific arrangement in the terms of the Centrifuge agreement, and may not apply to all of the RWA types. (These are new to all of us and we're working out the kinks)

If this is a common operation, we will likely be able to add a function to [dss-exec-lib](https://github.com/makerdao/dss-exec-lib) that handles this calculation based on the existing `mat`.